certauth.go

```diff
@@ -135,56 +182,35 @@ func (a *Auth) ValidateRequest(r *http.Request) error {
 	return nil
 }
 
-// Process is the main Entrypoint
-func (a *Auth) Process(w http.ResponseWriter, r *http.Request) error {
-	if err := a.ValidateRequest(r); err != nil {
-		return err
-	}
-
-	// Validate OU
-	if len(a.opt.AllowedOUs) > 0 {
-		err := a.ValidateOU(r.TLS.VerifiedChains[0][0])
-		if err != nil {
-			a.authErrHandler.ServeHTTP(w, r)
-			return err
+// CheckAuthorization runs each of the AuthorizationCheckers configured for the server
+// and returns an error if any of them return False.
+// See the documentation for AuthorizationChecker for more details.
+func (a *Auth) CheckAuthorization(
+	verifiedCert *x509.Certificate, ps httprouter.Params,
+) (map[ContextKey]ContextValue, error) {
+	ou := verifiedCert.Subject.OrganizationalUnit
+	cn := verifiedCert.Subject.CommonName
+
+	ctxParams := make(map[ContextKey]ContextValue)
+	var (
+		parms map[ContextKey]ContextValue
+		err   error
+	)
+
+	for _, ck := range a.opt.AuthorizationCheckers {
+		if ps == nil { // not using httprouter
+			parms, err = ck.CheckAuthorization(ou, cn)
+		} else { // using httprouter
```

Comment:
In this case that's not easy to do because `parms` is set here (or the else at 204) and then read again at 211.

And we want to make sure we only call one of `CheckAuthorization` or `CheckAuthorizationWithParams`, not both.